### PR TITLE
aarch64: a few trivial assembly optimizations

### PR DESCRIPTION
**Pull request by npitre**

### CODE REVIEW

**Comment by npitre** [comment]
split from #32594 as this stands on its own

**Review by carlocaione — APPROVED** [review]

File: arch/arm/core/aarch64/isr_wrapper.S [hunk 1]
```diff
@@ -83,8 +82,6 @@ spurious_continue:
 
 	/* --(_kernel->nested) */
 	dec_nest_counter x0, x1
-
-	cmp	x1, #0
 	bne	exit
```
Comment (jharris-intel) [inline arch/arm/core/aarch64/isr_wrapper.S:85]: Nit: I'd tend to prefer leaving `dec_nest_counter` as-is and changing these to `cbnz` instead. Same number of instructions and doesn't rely on carrying flags across (macro, admittedly) calls.

...though now that I look at it clang appears to prefer `subs` / `bne` here: https://godbolt.org/z/vrqzdd

File: arch/arm/core/aarch64/isr_wrapper.S [hunk 2]
```diff
@@ -83,8 +82,6 @@ spurious_continue:
 
 	/* --(_kernel->nested) */
```
Comment (jharris-intel) [inline arch/arm/core/aarch64/isr_wrapper.S:83]:
```suggestion
	/* if(!--(_kernel->nested)) { */
```

Nit: perhaps update the comment now that it's not nearly as clear what the `bne` below is doing?

**Review by jharris-intel — APPROVED** [review]

**Comment by npitre** [comment]
On Wed, 24 Feb 2021, jharris-intel wrote:

> I think it's ok; I want to check because this is a difference with the current implementation: clobbering x3 is ok here?

Yes. We are free to mess up with x0 to x18 here, and there is no 
intervening calls while the value in x3 has to be live.

> 	/* if(!--(_kernel->nested)) { */

Good point.

**Review by jharris-intel — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**